runelite-client/src/main/java/net/runelite/client/ui/ReducedFunctionalityPopup.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2018, Sam "Berry" Beresford <seb1g13@soton.ac.uk>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.ui;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Dialog;
+import javax.swing.JButton;
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.border.EmptyBorder;
+
+public class ReducedFunctionalityPopup extends JDialog
+{
+	public ReducedFunctionalityPopup()
```

Comment:
You can probably replace all of this with JOptionPane call.

Reply:
I could, but thought a custom dialog allowed better customisation.
Considering JOptionPanes can only have title, message and icon changed, whereas a dialog can have a custom layout.

Reply:
All you really want is the text anyway, and I believe there are plenty overloads to the JOptionPane methods that you can achieve what you want. Reinventing the wheel and creating completely different component from rest of the UI just for this 1 message is incorrect.

Reply:
Current Dialog:
![custompanel](https://user-images.githubusercontent.com/17437367/39411834-6f52b1f2-4c09-11e8-85ef-f9b5343be3fd.JPG)
JOptionPane:
![joptionpane](https://user-images.githubusercontent.com/17437367/39411835-71cfa85e-4c09-11e8-8ee0-393164f9be47.JPG)

Whats do you reckon?

Reply:
The JOptionPane layout does look much closer to the UI style used in the rest of the app. I'd go with that one, and probably space out the button by about one line.

Reply:
No I added it to see what it looks like, not too sure it though, what do you reckon?

Reply:
It does look quite friendly with the extra space. I'd just make sure the warning icon stays aligned with the top of the text.

Reply:
Don't think I can offset the icon in a JOptionPane, so have removed it for now
![image](https://user-images.githubusercontent.com/17437367/39412116-26968e60-4c0f-11e8-89ac-6765cadd98b3.png)

Reply:
Since the text relies on manual line breaks, I'd move the word `latest` to the first line so the word wrapping looks consistent.

Reply:
Managed to styled it with some bordering and padding:
![image](https://user-images.githubusercontent.com/17437367/39412857-4f7cb894-4c1a-11e8-9aa3-e9917d8a078b.png)